### PR TITLE
Feature/task 844 Extend the Budget Concept with a CostSummary

### PR DESCRIPTION
**Pull request by kempra**
* Adds the new categories CostSummary, CostSummaryEntry, CostType, and CostTypeCollection to the budget.cost concept.
* Supplies a new class CostSummaryEntriesCreator that creates the entries of the cost summary (1 entry per CostType)
* Supplies a button into the UI "Update Summary" that creates a table with 1 entry for each cost type in the structural tree
* Updated versions for cost concept plugins to match development versions for correct building

### CODE REVIEW

**Comment by SaMuellerDLR** [comment]
@kempra Has conflicts that need to be resolved.
Also check

https://github.com/virtualsatellite/VirtualSatellite4-Core/pull/904/checks?check_run_id=1669392983

for reported spotbugs issues.

File: de.dlr.sc.virsat.model.extension.budget.cost.ui/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -13,8 +13,13 @@ Require-Bundle: de.dlr.sc.virsat.model.extension.budget.cost,
...
  de.dlr.sc.virsat.project.ui,
- de.dlr.sc.virsat.uiengine.ui
+ de.dlr.sc.virsat.uiengine.ui,
+ org.eclipse.core.commands,
+ org.eclipse.emf.common
 Bundle-RequiredExecutionEnvironment: JavaSE-1.8
 Bundle-ActivationPolicy: lazy
 Automatic-Module-Name: de.dlr.sc.virsat.model.extension.budget.cost.ui
 Bundle-Vendor: DLR (German Aerospace Center)
+Import-Package: org.eclipse.gef.commands,
```
Comment (PhilMFischer) [inline de.dlr.sc.virsat.model.extension.budget.cost.ui/META-INF/MANIFEST.MF:23]: We should avoid usingimport package for consistency reasons.

**Review by PhilMFischer — CHANGES_REQUESTED** [review]
Some minor issue with using imported packages.

File: de.dlr.sc.virsat.model.extension.budget.cost.ui/META-INF/MANIFEST.MF (hunk 1 above)
Comment (SaMuellerDLR) [reply de.dlr.sc.virsat.model.extension.budget.cost.ui/META-INF/MANIFEST.MF]: Nice catch 👍 Cleaned up the import statements.

**Review by PhilMFischer — APPROVED** [review]

**Pull request merged**